DOC Improve group based CV splitter docs (#23861)
scikit-learn/scikit-learn: doc/modules/cross_validation.rst

Each subject is in a different testing fold, and the same subject is never in
both testing and training. Notice that the folds do not have exactly the same

> size due to the imbalance in the data. If balanced classes are desired,
> :class:`StratifiedGroupKFold` may be a better option.

Reviewer: Is this more exact?
```suggestion
size due to the imbalance in the data. If classes proportions must be
balanced across folds, :class:`StratifiedGroupKFold` is a better option.
```

   :align: center
   :scale: 75%

Similar to k-fold, the test sets from :class:`GroupKFold` will form a complete
partition of all the data. Unlike k-fold, :class:`GroupKFold` is not randomized

> at all, whereas :class:`KFold` is randomized when ``shuffle=True``.

Reviewer: To me, the distinction between :class:`KFold` and `"k-fold"` is not really clear. Can this be made more precise?

(To me, :class:`KFold` is the simplest strategy of k-folding.)

Author: Ah I thought they meant the same thing, I have amended to use :class:Kfold instead

> Similar to :class:`KFold`, the test sets from :class:`GroupKFold` will form a
> complete partition of all the data. Unlike k-fold, :class:`GroupKFold` is not
> randomized at all, whereas :class:`KFold` is randomized when ``shuffle=True``.

Reviewer: ```suggestion
Similar to :class:`KFold`, the test sets from :class:`GroupKFold` will form a
complete partition of all the data. Unlike :class:`KFold`, :class:`GroupKFold`
is not randomized at all, whereas :class:`KFold` is randomized when
``shuffle=True``.
```

cross-validation folds.

Each training set is thus constituted by all the samples except the ones
related to a specific group. This is basically the same as :class:`GroupKFold`

> with ``k=1``.

Author: Yes that makes much more sense

^^^^^^^^^^^^^^^^^^

:class:`LeavePGroupsOut` is similar as :class:`LeaveOneGroupOut`, but removes
samples related to :math:`P` groups for each training/test set. All possible
combinations of :math:`P` groups are left out, meaning test sets will overlap

> for :math:`P>1`.

Author: Yes, I just thought the original was confusing and wanted to make it clearer. Happy to change it back?

> for :math:`P>1`.

Reviewer 2: I'm okay to change it back.

> for :math:`P>1`.

Reviewer 2: Are you planning to revert this change?

> for :math:`P>1`.

Author: I changed to what you suggested? Or did I mis-understand?

> for :math:`P>1`.

Reviewer 2: From https://github.com/scikit-learn/scikit-learn/pull/23861#discussion_r924088971, I thought you suggested to revert the whole sentence.

Looking at this again, I am okay with the current wording.
